builtin/providers/aws/resource_aws_api_gateway_usage_plan.go

```diff
@@ -0,0 +1,498 @@
+		Delete: resourceAwsApiGatewayUsagePlanDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true, // Required since not addable nor removable afterwards
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"api_stages": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"api_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"stage": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"quota_settings": {
+				Type:     schema.TypeSet,
+				MaxItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"limit": {
+							Type:     schema.TypeInt,
+							Required: true, // Required as not removable singularly
+						},
+
+						"offset": {
+							Type:     schema.TypeInt,
+							Default:  0,
+							Optional: true,
+						},
+
+						"period": {
+							Type:         schema.TypeString,
+							Required:     true, // Required as not removable
+							ValidateFunc: validateApiGatewayUsagePlanQuotaSettingsPeriod,
+						},
+					},
+				},
+			},
+
+			"throttle_settings": {
+				Type:     schema.TypeSet,
+				MaxItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"burst_limit": {
+							Type:     schema.TypeInt,
+							Default:  0,
+							Optional: true,
+						},
+
+						"rate_limit": {
+							Type:     schema.TypeInt,
+							Default:  0,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"product_code": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayUsagePlanCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+	log.Print("[DEBUG] Creating API Gateway Usage Plan")
+
+	params := &apigateway.CreateUsagePlanInput{
+		Name: aws.String(d.Get("name").(string)),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		params.Description = aws.String(v.(string))
+	}
+
+	if s, ok := d.GetOk("api_stages"); ok {
+		stages := s.([]interface{})
+		as := make([]*apigateway.ApiStage, 0)
+
+		for _, v := range stages {
+			sv := v.(map[string]interface{})
+			stage := &apigateway.ApiStage{}
+
+			if v, ok := sv["api_id"].(string); ok && v != "" {
+				stage.ApiId = aws.String(v)
+			}
+
+			if v, ok := sv["stage"].(string); ok && v != "" {
+				stage.Stage = aws.String(v)
+			}
+
+			as = append(as, stage)
+		}
+
+		if len(as) > 0 {
+			params.ApiStages = as
+		}
+	}
+
+	if v, ok := d.GetOk("quota_settings"); ok {
+		settings := v.(*schema.Set).List()
+		q, ok := settings[0].(map[string]interface{})
+
+		if errors := validateApiGatewayUsagePlanQuotaSettings(q); len(errors) > 0 {
```

Comment:
Do you mean setting the ValidateFunc? Well, I need to figure out the combination of 2 fields, and don't know how to proceed better :/